docs/release-notes/flink-1.12.md

```diff
@@ -0,0 +1,193 @@
+
+
+#### Upgrade to Avro version 1.10.0 from 1.8.2 [FLINK-18192](https://issues.apache.org/jira/browse/FLINK-18192)
+
+The default version of Avro in flink-avro module was upgraded to 1.10. If for some reason you need an older version (you have Avro coming from Hadoop, or you use classes generated from an older Avro version), please explicitly downgrade the Avro version in your project. 
+
+NOTE: We observed a decreased performance of the Avro 1.10 version compared to 1.8.2. If you are concerned with the performance and you are fine working with an older version of Avro, consider downgrading the Avro version. 
+
+
+#### Create an uber jar when packaging flink-avro for sql client [FLINK-18802](https://issues.apache.org/jira/browse/FLINK-18802)
+
+The sql client jar was renamed to flink-sql-avro-{{site.version}}.jar, previously flink-avro-{{site.version}}-sql-jar.jar. Moreover it is no longer needed to add Avro dependencies manually. 
+
+
+### Deployment
+
+#### Enable rolling in default log settings of standalone scripts [FLINK-8357](https://issues.apache.org/jira/browse/FLINK-8357)
+
+The default log4j configuration has changed: Besides the existing rolling of log files on startup of Flink, they also roll once they've reached a size of 100MB. Flink keeps a total of 10 log files, effectively limiting the total size of the log directory to 1GB (per Flink service logging to that directory).
+
+
+#### Use jemalloc by default in the Flink docker image [FLINK-19125](https://issues.apache.org/jira/browse/FLINK-19125)
+
+jemalloc is adopted as the default memory allocator in Flink's docker image to reduce issues with memory fragmentation. Users can roll back to using glibc by passing the 'disable-jemalloc' flag to the `docker-entrypoint.sh` script. For more details, please refer to Flink's documentation. 
+
+
+#### Upgrade Mesos version to 1.7 [FLINK-19783](https://issues.apache.org/jira/browse/FLINK-19783)
+
+The Mesos dependency has been bumped from 1.0.1 to 1.7.0. 
+
+
+#### Send SIGKILL if Flink process doesn't stop after a timeout [FLINK-17470](https://issues.apache.org/jira/browse/FLINK-17470)
+
+In Flink 1.12 we changed the behavior of the standalone scripts to issue a SIGKILL if a SIGTERM did not succeed in shutting down a Flink process. 
+
+
+#### Introduce non-blocking job submission [FLINK-16866](https://issues.apache.org/jira/browse/FLINK-16866)
+
+The semantics of submitting a job have slightly changed. The submission call returns almost immediately, with the job being in a new INITIALIZING state. Operations such as triggering a savepoint or retrieving the full job details are not available while the job is in that state. 
+Once the JobManager for that job has been created, the job is in `CREATED` state and all calls are available. 
+
+
+### Runtime
```

Comment:
Shouldn't we add also new features like the UI update of [FLIP-102](https://cwiki.apache.org/confluence/display/FLINK/FLIP-102%3A+Add+More+Metrics+to+TaskManager) and [FLIP-104](https://cwiki.apache.org/confluence/display/FLINK/FLIP-104%3A+Add+More+Metrics+to+Jobmanager) to the release notes? If not, we, at least, should mention the newly introduced metrics for managed memory, network memory, and metaspace as we had questions about them popping up in the mailing list.

Reply:
I believe new features should be announced in the Release Announcement Blog Post, ideally with a link to the updated documentation.
These release notes are focusing on changes in APIs, semantics, configuration options etc.